### PR TITLE
update-python-libraries: use sri-hash

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
lots of conflict pain around https://github.com/NixOS/nixpkgs/pull/154969 where the update script would use different shas from the rest of the ecosystem.

This is to make the sha's reflect what someone would manually produce

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
did test this with hypothesis

**Review by FRidh — APPROVED** [review]

File: pkgs/development/interpreters/python/update-python-libraries/update-python-libraries.py [hunk 1]
```diff
@@ -340,13 +340,14 @@ def _update_package(path, target):
         raise ValueError("no file available for {}.".format(pname))
 
     text = _replace_value('version', new_version, text)
+    # hashes from pypi are 16-bit encoded sha256's, need translate to an sri hash if used with "hash"
+    # this is the default format for nix since 2.4
+    sri_hash = subprocess.check_output(["nix", "hash", "to-sri", "--type", "sha256", new_sha256]).decode('utf-8').strip()
```
Comment (mweinelt) [inline pkgs/development/interpreters/python/update-python-libraries/update-python-libraries.py:345]:
```suggestion
    try:
        sri_hash = subprocess.check_output(["nix", "hash", "to-sri", "--type", "sha256", new_sha256]).decode('utf-8').strip()
    except subprocess.CalledProcessError:
        # nix<2.4 compat
        sri_hash = subprocess.check_output(["nix", "to-sri", "--type", "sha256", "new_sha256]).decode('utf-8').strip()
        
```

```
❯ nix hash
error: 'hash' is not a recognised command
❯ nix to-sri --help
Usage: nix to-sri <FLAGS>... <STRINGS>...

Summary: convert a hash to SRI representation.

Flags:
      --type <TYPE>  hash algorithm ('md5', 'sha1', 'sha256', or 'sha512')

Note: this program is EXPERIMENTAL and subject to change.
```

**Comment by mweinelt** [comment]
>	350: Trailing whitespace

:smile:

**Comment by jonringer** [comment]
yea, was going to edit the comments as well

**Comment by jonringer** [comment]
done

**Pull request merged**